soc/ti/k3/am6x/Kconfig.soc

```diff
@@ -29,9 +35,14 @@ config SOC_AM6442_M4
 	bool
 	select SOC_SERIES_AM6X_M4
 
+config SOC_J721E_MAIN_R5F0_0
+	bool
+	select SOC_SERIES_AM6X_R5
+
 config SOC_SERIES
 	default "am6x" if SOC_SERIES_AM6X
 
 config SOC
 	default "am6234" if SOC_AM6234_M4 || SOC_AM6234_A53
 	default "am6442" if SOC_AM6442_M4
+	default "j721e" if SOC_J721E_MAIN_R5F0_0
```

Comment:
This seems inconsistent with soc/ti/k3/am6x/Kconfig SOC_PART_NUMBER. Should it be leading caps, all caps or lower-case? Why would the SOC_PART_NUMBER be different than SOC?

Reply:
For `SOC` I found the following in the porting guide:

```
config SOC
    default "SoC name" if SOC_<SOC_NAME>

Notice that ``SOC_NAME`` is a pure upper case version of the SoC name.

Notice that the string value must match the SoC name used in the :file:`soc.yml` file.
```

To me that seems to indicate all lower to match what we do in soc.yml.

For `SOC_PART_NUMBER`, to be honest I have no clue, I don't see it used anywhere either, I just added it based on what the other TI SoCs had done, but maybe they didn't need it either?

Reply:
`SOC` must be lower, `SOC_PART_NUMBER` does not exist, i.e. it's a custom field, it has no relation to anything and is not used by anything, it can be safely removed unless you have HAL code using it